clients/cloud/clojure/README.md

```diff
@@ -0,0 +1,79 @@
+# Overview
+
+Produce messages to and consume messages from [Confluent
+Cloud](https://www.confluent.io/confluent-cloud/) using Java interop
+form Clojure.
+
+For more information, please see the [application development documentation](https://docs.confluent.io/current/api-javadoc.html)
+
+# Prerequisites
+
+* Java 8 or higher (Clojure 1.10 recommends using Java 8 or Java 11)
+* The [Leiningen](https://leiningen.org/#install) tool to compile and run the demos
+* Access to a [Confluent Cloud](https://www.confluent.io/confluent-cloud/) cluster
+* Local file with configuration parameters to connect to your Confluent Cloud instance ([how do I find those?](https://docs.confluent.io/current/cloud/using/config-client.html#librdkafka-based-c-clients)). Format the file as follows:
+
+
+```bash
+$ cat ~/.ccloud/example.config
+bootstrap.servers=<broker-1,broker-2,broker-3>
+sasl.username=<api-key-id>
+sasl.password=<secret-access-key>
+ssl.endpoint.identification.algorithm=https
+security.protocol=SASL_SSL
+sasl.mechanism=PLAIN
+```
+
+# Example 1: Hello World!
+
+In this example, the producer writes Kafka data to a topic in Confluent Cloud.
+Each record has a key representing a username (e.g. `alice`) and a value of a count, formatted as json (e.g. `{"count": 0}`).
+The consumer reads the same topic from Confluent Cloud and keeps a rolling sum of the counts as it processes each record.
+
+1. Run the producer, passing in arguments for (a) the local file with configuration parameters to connect to your Confluent Cloud instance and (b) the topic name:
+
+```shell
+$ lein producer ~/.ccloud/example.config test1
+…
+Producing record: alice 	{"count":0}
+Producing record: alice 	{"count":1}
+Producing record: alice 	{"count":2}
+Producing record: alice 	{"count":3}
+Producing record: alice 	{"count":4}
+Produced record to topic test1 partiton [0] @ offest 0
+Produced record to topic test1 partiton [0] @ offest 1
+Produced record to topic test1 partiton [0] @ offest 2
+Produced record to topic test1 partiton [0] @ offest 3
+Produced record to topic test1 partiton [0] @ offest 4
+Producing record: alice 	{"count":5}
+Producing record: alice 	{"count":6}
```

Comment:
(this differs from the Python example because there are two blocks of messages: one being sent with Callback, one by derefing the future, to demonstrate both methods)